[Doc] runtime env FAQ optimization (#25008)
ray-project/ray: doc/source/ray-core/handling-dependencies.rst

"""""""""""""""""""""""""""""""""""""""""

If a runtime environment is specified in ``ray.init(runtime_env=...)``, then the environment will be installed on every node.  See :ref:`Per-Job <rte-per-job>` for more details.

> If you don't want to install the environment on every node, add ``"eager_install": False`` to the ``runtime_env``. It means that the environment will be installed on demand on the nodes which your tasks or actors are scheduled to.

Reviewer: ```suggestion
Note, by default the runtime environment will be installed eagerly on every node in the cluster. If you want to lazily install the runtime environment on demand, use set the ``eager_install`` option to false: ``ray.init(runtime_env={..., "eager_install": True}``. 
```

> (Note, by default the runtime environment will be installed eagerly on every node in the cluster. If you want to lazily install the runtime environment on demand, set the ``eager_install`` option to false: ``ray.init(runtime_env={..., "eager_install": True}``.)

Reviewer 2: ```suggestion
(Note, by default the runtime environment will be installed eagerly on every node in the cluster. If you want to lazily install the runtime environment on demand, set the ``eager_install`` option to false: ``ray.init(runtime_env={..., "eager_install": False}``.)
```

Reviewer 2: How about we leave this in the `When is the environment installed?` section and add the detail there instead? Seems like the most logical place instead of duplicating the information around

Author: My concern is that maybe some users only read the section `Are environments installed on every node?` and they may misunderstand that the job level runtime_env can only be installed on all nodes eagerly. Wdyt?

Reviewer 2: Actually, I think as-is is good otherwise it's less of a FAQ...

When is the environment installed?
""""""""""""""""""""""""""""""""""

> When specified per-job, the environment is installed when you call ``ray.init()`` unless that ``"eager_install": False`` is set.

Reviewer 2: ```suggestion
When specified per-job, the environment is installed when you call ``ray.init()`` (unless ``"eager_install": False`` is set).
```

On the other hand, loading a runtime environment from the cache should be nearly as fast as the ordinary Ray worker startup time, which is on the order of a few seconds. A new Ray worker is started for every Ray actor or task that requires a new runtime environment.
(Note that loading a cached ``conda`` environment could still be slow, since the ``conda activate`` command sometimes takes a few seconds.)

> You can set ``setup_timeout_seconds`` config to avoid the installation hanging for a long time.

Reviewer: Should also say what the behavior is when timing out?

> You can set ``setup_timeout_seconds`` config to avoid the installation hanging for a long time. If the installation is not finished within this time, your tasks or actors will fail.

Reviewer 2: ```suggestion
You can set ``setup_timeout_seconds`` config to avoid the installation hanging for a long time. If the installation is not finished within this time, your tasks or actors will fail to start.
```
